org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/jobs/EclipseInputFile.java

```diff
@@ -127,4 +130,8 @@ public InputStream inputStream() throws IOException {
     return new ByteArrayInputStream(contents().getBytes(getCharset()));
   }
 
+  public boolean isOlderThan(IDocument document) {
+    return documentModificationStamp < ((IDocumentExtension4) document).getModificationStamp();
```

Comment:
Does it work for closed editors? What is the modificationStamp of a Document created using the `ITextFileBufferManager`?

Reply:
Nope, that's why a test fails at the moment. Will try to fix that